Mathlib/Analysis/Distribution/SchwartzSpace.lean

```diff
@@ -1003,17 +1017,85 @@ theorem iteratedPDeriv_eq_iteratedFDeriv {n : ℕ} {m : Fin n → E} {f : 𝓢(E
 
 end Derivatives
 
+section Integration
+
+/-! ### Integration -/
+
+
+open Real Complex Filter MeasureTheory MeasureTheory.Measure FiniteDimensional
+
+variable [NormedAddCommGroup D] [NormedSpace ℝ D] [NormedAddCommGroup V] [NormedSpace ℂ V]
+variable [MeasurableSpace D] [BorelSpace D] [FiniteDimensional ℝ D]
+
+lemma integrable_pow_mul (f : 𝓢(D, V)) {μ : Measure D} (k : ℕ) [IsAddHaarMeasure μ] :
+    Integrable (fun x ↦ ‖x‖ ^ k * ‖f x‖) μ := by
+  let l := finrank ℝ D + 1 + k
+  obtain ⟨C, C_nonneg, hC⟩ : ∃ C, 0 ≤ C ∧ ∀ x, (1 + ‖x‖) ^ l * ‖f x‖ ≤ C := by
+    use 2 ^ l * (Finset.Iic (l, 0)).sup (fun m ↦ SchwartzMap.seminorm ℝ m.1 m.2) f, by positivity
+    simpa using f.one_add_le_sup_seminorm_apply (m := (l, 0)) (k := l) (n := 0) le_rfl le_rfl
+  have : Integrable (fun x : D ↦ C * (1 + ‖x‖) ^ (-((finrank ℝ D + 1 : ℕ) : ℝ))) μ := by
+    apply (integrable_one_add_norm ?_).const_mul
+    exact Nat.cast_lt.mpr Nat.le.refl
+  apply this.mono ((aestronglyMeasurable_id.norm.pow _).mul f.continuous.aestronglyMeasurable.norm)
+    (eventually_of_forall (fun x ↦ ?_))
+  conv_rhs => rw [norm_of_nonneg (by positivity), rpow_neg (by positivity), ← div_eq_mul_inv]
+  rw [le_div_iff' (by positivity)]
+  simp only [id_eq, Pi.mul_apply, Pi.pow_apply, norm_mul, norm_pow, norm_norm, rpow_nat_cast]
+  calc
+    (1 + ‖x‖) ^ (finrank ℝ D + 1) * (‖x‖ ^ k * ‖f x‖)
+      ≤ (1 + ‖x‖) ^ (finrank ℝ D + 1) * ((1 + ‖x‖) ^ k * ‖f x‖) := by gcongr; simp
+    _ = (1 + ‖x‖) ^ (finrank ℝ D + 1 + k) * ‖f x‖ := by simp only [pow_add, mul_assoc]
+    _ ≤ C := hC x
+
+lemma integrable (f : 𝓢(D, V)) {μ : Measure D} [IsAddHaarMeasure μ] :
+    Integrable f μ :=
+  (f.integrable_pow_mul (μ := μ) 0).mono f.continuous.aestronglyMeasurable
+    (eventually_of_forall (fun _ ↦ by simp))
+
+/-- The integral as a continuous linear map from Schwartz space to the codomain. -/
+def integral (μ : Measure D) [IsAddHaarMeasure μ] : 𝓢(D, V) →L[ℝ] V :=
```

Comment:
Maybe this should be called `integralCLM`, for consistency with other similar declarations?

Reply:
I would rather remove some of the 'CLM's in names where it is no longer necessary.

Reply:
What does "necessary" mean, though? It's a pretty widespread convention in mathlib that when you're wrapping up some relatively simple operation into a more complicated homomorphism class, this is indicated with a suffix to the name indicating what you're packaging it as; e.g. `Nat.castRingHom`, or `ContinuousMap.evalCLM`. I'd definitely be happier if this were adhered to here. (We can raise this one on zulip if you like.)

Reply:
By necessary I mean that there is a conflicting definition for the bare function. I don't have too strong opinions on that and if you prefer `integralCLM` I can rename it.

Reply:
But there is a conflict, isn't there? If you have `SchwartzMap` and `MeasureTheory` open at the same time (not a totally unlikely prospect) then you get a conflict between `SchwartzMap.integral` and `MeasureTheory.integral`.